### ISSUE TITLE
[android] when used as material-top-tab with react-navigation, navigation.navigate may not complete screen transition

### ISSUE THREAD

**Issue by lovegaoshi** [callstack/react-native-pager-view#910]
## Environment

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 7.0.1
| @react-navigation/drawer               | 7.0.1
| @react-navigation/material-top-tabs    | 7.0.1
| react-native-pager-view                |6.5.0
| react-native                           |0.76.1
| expo                                   |20.0.2
| node                                   | 20
| npm or yarn                            | 4

## Description

hi! crossposted from https://github.com/react-navigation/react-navigation/issues/12251
I use a custom drawer content to put buttons that navigate to specific screens as material top tabs, which are nested as a drawer screen. since the new arch migration, I observed the sliding transition of the tab screens are 1. too fast; 2. it occasionally does not do a full transition in dev, and almost never in release. 
this behavior is not observed in the old arch.
Many thanks for anyone having a look at this!

see the attached video; error occurs toward the very end
https://github.com/user-attachments/assets/42b7e0b4-d784-4ae4-9595-99926aa0126b

## Reproducible Demo

My reproducer is adapted from my app, but I trimmed all of my code and left only Bug.tsx as the actual reproducer. 
https://github.com/lovegaoshi/azusa-player-mobile/tree/bug

**Comment by infotheme-dev** [comment]
Hi all I found a solution and it's woking like charm look you have to follow one thing here : https://github.com/callstack/react-native-pager-view/pull/907/files

 "react-native-gesture-handler": "2.20.2",
    "react-native-reanimated": "3.16.1",
    "react-native-safe-area-context": "4.14.0",
    **"react-native-screens": "4.0.0",**  and this 4.0.0
    "react-native-svg": "15.8.0",
    **"react-native-tab-view": "^3.3.0"**  Keep the version of tab view 3.3.0 and 

Check you package.json and install these libraries like above mentioned and you will see the issue has been resolved 

react-native-pager-view latest version

**Comment by infotheme-dev** [comment]
Hi all I found a solution and it's woking like charm look you have to follow one thing here : simply disable newArchEnabled=false in gradle properties, This is a problem with newArch

**Comment by lovegaoshi** [comment]
thanks, but im actively trying to migrate to newarch as well, since expo
"threatened" dropping old arch support in 2025.

**Comment by infotheme-dev** [comment]
Yes this is for those who are developing things right now, So that they can continue with dev. Did you tried to do a workaround using React Native's backHandler to return false;

**Comment by lovegaoshi** [comment]
pardon me im not aware of how to put that backhandler part; i thought the
bug is navigation. navigate into a nested screen, i have no problem sliding
or using the back gesture
